### ISSUE TITLE
Return empty list if ListField has no items?

### ISSUE THREAD

**Issue by spro** [hmarr/mongoengine#53]
Rather than returning None, which can't be worked with much, maybe return an empty list if a Document's ListField is empty/isn't set? So you could have something like this Document subclass:

```
class Post(Document):
    comments = ListField(...) # but this isn't set yet
    ...
```

 and still do something like `len(post.comments)` without getting an exception thrown.

**Comment by justquick** [comment]
+1

**Comment by flosch** [comment]
I commited that in http://github.com/hmarr/mongoengine/commit/6093e88eebf932d04741364f53de77878198c1f1 

Same for the DictField where it is now an empty {} by default. 

Fixed, please close this issue.

**Comment by hmarr** [comment]
This is fixed in 0.4
